binary-quantization-mistral/bq-mistral-embed.md

```diff
@@ -0,0 +1,537 @@
+
+```python
+client.recreate_collection(
+    collection_name=collection_name,
+    vectors_config=models.VectorParams(
+        size=1024,
+        distance=models.Distance.COSINE,
+    ),
+    optimizers_config=models.OptimizersConfigDiff(
+        default_segment_number=5,
+        indexing_threshold=0,
+    ),
+    quantization_config=models.BinaryQuantization(
+        binary=models.BinaryQuantizationConfig(always_ram=True),
+    ),
+    shard_number=2,
+)
+```
+
+### Checking and Populating the Collection
+
+In this process, the initial step involves verifying whether the collection within Qdrant, specified by `collection_name`, currently contains any vectors. The existence of vectors within a collection is a crucial aspect as it signifies whether the collection is ready for operations such as searches or requires population.
+
+The code snippet utilizes the `get_collection` method provided by the Qdrant client to fetch details about the specific collection. It then checks if the `vectors_count` attribute of the collection info is zero, indicating the collection is empty and requires population.
+
+To populate the collection, the code employs an effective batching strategy for the upsert operation. Here, the variable `bs` represents the batch size, set to 1000, defining how many points will be inserted into the collection in each operation. This approach reduces the load on the database and prevents potential timeouts or performance degradation that might occur when trying to insert large volumes of data in a single operation.
+
+Through a loop, slices of points, each of size `bs`, are created and inserted (upserted) into the collection using the `upsert` method of the Qdrant client. This process is visually represented using a progress bar, which completes once all points are successfully upserted into the collection.
+
+The output confirms the initial condition—the collection was indeed empty and the upsert operation began. The logged information and the completion of the progress bar signify not only the start but also the successful end of the operation, with all batches of points inserted into the collection. This step is fundamental for subsequent operations, such as querying or searching the collection, as it ensures that there's data within the collection to act upon.
+
+```python
+collection_info = client.get_collection(collection_name=collection_name)
+
+if collection_info.vectors_count == 0:
+    logger.info("Collection is empty. Begin upsert.")
+    bs = 1000  # Batch size
+    for i in tqdm(range(0, len(points), bs)):
+        slice_points = points[i : i + bs]  # Create a slice of bs points
+        client.upsert(collection_name=collection_name, points=slice_points)
```

Comment:
I would appreciate using `upload_collection` or `upload_points` (with `parallel != 1` if possible, but it might not be applicable no jupyter notebooks)

However, it is also good since demonstrates a need to do batching before sending large amount of data.